lib/v2/layoffs/index.js

```diff
@@ -57,12 +44,59 @@ module.exports = async (ctx) => {
         'x-time-zone': 'America/Los_Angeles',
     };
 
+    // Get the latest data source url
+    let dataSourceUrl = await ctx.cache.get(ENTRY_URL);
+    let cacheInvalid = false;
+    let response;
+    if (!dataSourceUrl) {
+        cacheInvalid = true;
+    } else {
+        // Try to fetch data,
+        // it may fail due to outdated url
+        try {
+            response = await got({
+                method: 'get',
+                url: dataSourceUrl,
+                headers,
+            });
+            if (!response.ok) {
+                cacheInvalid = true;
+            }
+        } catch {
+            cacheInvalid = true;
+        }
+    }
+
+    if (cacheInvalid) {
+        // Refetch the data source link from entry page
+        const sourcePage = await got({
+            method: 'get',
+            url: ENTRY_URL,
+        });
+        const $ = cheerio.load(sourcePage.data);
+        const SEARCH_STRING = 'urlWithParams: "';
+        for (const scriptObj of $('script').get()) {
+            const scriptText = $(scriptObj).text();
+            const index = scriptText.indexOf(SEARCH_STRING);
+            if (index > 0) {
+                const nextIndex = scriptText.indexOf('"', index + SEARCH_STRING.length);
+                dataSourceUrl = AIRTABLE_HOST + scriptText.substring(index + SEARCH_STRING.length, nextIndex).replaceAll('\\u002F', '/');
+                break;
+            }
+        }
```

Comment:
```suggestion
        dataSourceUrl =
            AIRTABLE_HOST +
            $('script')
                .text()
                .match(/urlWithParams: "(.*?)"/)[1]
                .replace(/\\u002F/g, '/');
```
Note that RSSHub supports down to node v14 which does not have `replaceAll`

Reply:
Thanks for the more elegant solution!